docs/docs/features/soft-off.md

```diff
@@ -0,0 +1,255 @@
+
+#### Soft Off Waker
+
+Next, we need to add another device which will be enabled only when the keyboard is going into soft off state, and will configure the previously declared GPIO key with the correct interrupt configuration to wake the device from soft off once it is pressed.
+
+```
+/ {
+    wakeup_source: wakeup_source {
+        compatible = "zmk,gpio-key-wakeup-trigger";
+
+        trigger = <&wakeup_key>;
+        wakeup-source;
+    };
+};
+```
+
+Here are the properties for the node:
+
+- The `compatible` property for the node must be `zmk,gpio-key-wakeup-trigger`.
+- The `trigger` property is a phandle to the GPIO key defined earlier.
+- The `wakeup-source` property signals to Zephyr this device should not be suspended during the shutdown procedure.
+- An optional `output-gpios` property contains a list of GPIO pins (including the appropriate flags) to set active before going into power off, if needed to ensure the GPIO pin will trigger properly to wake the keyboard. This is only needed for matrix integrated combos. For those keyboards, the list should include the matrix output needs needed so the combo hardware is properly "driven" when the keyboard is off.
+
+Once that is declared, we will list it in an additional configuration section so that the ZMK soft off process knows it needs to enable this device as part of the soft off processing:
+
+```
+/ {
+    soft_off_wakers {
+        compatible = "zmk,soft-off-wakeup-sources";
+        wakeup-sources = <&wakeup_source>;
+    };
+};
+```
+
+Here are the properties for the node:
+
+- The `compatible` property for the node must be `zmk,soft-off-wakeup-sources`.
+- The `wakeup-sources` property is a [phandle array](../config/index.md#devicetree-property-types) pointing to all the devices that should be enabled during the shutdown process to be sure they can later wake the keyboard.
+
+#### Soft Off Behavior Instance
+
+To use the [soft off behavior](../behaviors/soft-off.md) outside of a keymap, add an instance of the behavior to your `.overlay`/`.dts` file:
+
+```
+/ {
+    behaviors {
+        hw_soft_off: hw_soft_off {
+            compatible = "zmk,behavior-soft-off";
+            #binding-cells = <0>;
+            hold-time-ms = <5000>;
+        };
+    };
+};
+```
+
+#### KScan Sideband Behavior
+
+The kscan sideband behavior driver will be used to trigger the [soft off behavior](../behaviors/soft-off.md) "out of band" from the normal keymap processing. To do so, it will decorate/wrap an underlying kscan driver. What kscan driver will vary for simple direct pin vs. matrix-integrated hardware combo.
+
+##### Simple Direct Pin
+
+With a simple direct pin setup, the The [direct kscan](../config/kscan.md) driver can be used with a GPIO key, to make a small "side matrix":
+
+```
+    soft_off_direct_scan: soft_off_direct_scan {
+        compatible = "zmk,kscan-gpio-direct";
+        input-keys = <&wakeup_key>;
```

Comment:
Can this also not be simplified to get rid of `input-keys` (and `gpio-keys` usage altogether) by asking the user to augment `input-gpios` instead? That seems to be the approach already for matrix input.

Reply:
If we are using `input-keys`, it needs to be documented in the direct kscan config page.

Reply:
> Can this also not be simplified to get rid of `input-keys` (and `gpio-keys` usage altogether) by asking the user to augment `input-gpios` instead? That seems to be the approach already for matrix input.

The advantage is avoiding duplicating the GPIO details if we can avoid it, since we use the GPIO key for the waker elsewhere too. Unfortunately, we can't easily "mix" these in a matrix definition.

Reply:
I think keeping it conceptually simple would have been an advantage but if you prefer it this way then I am OK with that.

Reply:
I think we stick with it. It gives us some flexibility, especially for designs that already have the GPIO keys defined for the upstream board definition, etc.

Added a note to the kscan config about this.

Thoughts?

Reply:
Good call to note it there, sounds good to me.